src/components/AppProvider/AppProvider.tsx

```diff
@@ -45,89 +40,52 @@ export interface AppProviderProps extends AppBridgeOptions {
   UNSTABLE_telemetry?: TelemetryObject;
 }
 
-export class AppProvider extends React.Component<AppProviderProps, State> {
-  private stickyManager: StickyManager;
-  private scrollLockManager: ScrollLockManager;
-  private uniqueIdFactory: UniqueIdFactory;
-
-  constructor(props: AppProviderProps) {
-    super(props);
-    this.stickyManager = new StickyManager();
-    this.scrollLockManager = new ScrollLockManager();
-    this.uniqueIdFactory = new UniqueIdFactory(globalIdGeneratorFactory);
-
-    const {i18n, apiKey, shopOrigin, forceRedirect, linkComponent} = this.props;
-
-    // eslint-disable-next-line react/state-in-constructor
-    this.state = {
-      link: linkComponent,
-      intl: new I18n(i18n),
-      appBridge: createAppBridge({shopOrigin, apiKey, forceRedirect}),
-    };
-  }
-
-  componentDidMount() {
-    if (document != null) {
-      this.stickyManager.setContainer(document);
-    }
-  }
-
-  componentDidUpdate({
-    i18n: prevI18n,
-    linkComponent: prevLinkComponent,
-    apiKey: prevApiKey,
-    shopOrigin: prevShopOrigin,
-    forceRedirect: prevForceRedirect,
-  }: AppProviderProps) {
-    const {i18n, linkComponent, apiKey, shopOrigin, forceRedirect} = this.props;
-
-    if (
-      i18n === prevI18n &&
-      linkComponent === prevLinkComponent &&
-      apiKey === prevApiKey &&
-      shopOrigin === prevShopOrigin &&
-      forceRedirect === prevForceRedirect
-    ) {
-      return;
-    }
+export function AppProvider({
+  apiKey,
+  shopOrigin,
+  forceRedirect,
+  i18n: i18nProp,
+  linkComponent,
+  theme = {},
+  features = {},
+  UNSTABLE_telemetry,
+  children,
+}: AppProviderProps) {
+  const stickyManager = useLazyRef(() => new StickyManager());
+  const scrollLockManager = useLazyRef(() => new ScrollLockManager());
+  const uniqueIdFactory = useLazyRef(
+    () => new UniqueIdFactory(globalIdGeneratorFactory),
+  );
 
-    // eslint-disable-next-line react/no-did-update-set-state
-    this.setState({
-      link: linkComponent,
-      intl: new I18n(i18n),
-      appBridge: createAppBridge({shopOrigin, apiKey, forceRedirect}),
-    });
-  }
+  const appBridge = useMemo(
+    () => createAppBridge({shopOrigin, apiKey, forceRedirect}),
+    [apiKey, shopOrigin, forceRedirect],
+  );
+  const i18n = useMemo(() => new I18n(i18nProp), [i18nProp]);
 
-  render() {
-    const {
-      theme = {},
-      features = {},
-      UNSTABLE_telemetry,
-      children,
-    } = this.props;
-    const {intl, appBridge, link} = this.state;
+  useEffect(() => {
+    stickyManager.current.setContainer(document);
+  }, [stickyManager]);
```

Comment:
Should this be `stickyManager.current`? I can never remember how we want to handle effects based on refs

Reply:
As yes, nice catch. Assuming we want to re-run when the stickyManager changes we would want [ref.current]. However, this was auto-filled because we're using a custom hook. We should be able to omit. I'll omit it and re- 🎩 in web

Reply:
Removed the auto ignore and added a comment about why we wouldn't want to use ref's as a dependency